tests/app/core/models/session-model.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { assertSessionDataIntegrity } from '../../../../src/app/core/models/session.model';
+import { of } from 'rxjs';
+
+const validSessionData = {
+  openPhases: [
+    'phaseBugReporting',
+    'phaseTeamResponse',
+    'phaseTesterResponse',
+    'phaseModeration',
+  ],
+  phaseBugReporting: 'bugreporting',
+  phaseTeamResponse: 'pe-results',
+  phaseTesterResponse: 'testerresponse',
+  phaseModeration: 'pe-evaluation',
+};
+
+describe('Session Model', () => {
+  describe('Assert Session Data Integrity', () => {
+    it('should throw error on unavailable session', () => {
+      of(undefined)
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(new Error('Session Data Unavailable')),
+        });
+    });
+    it('should throw error on incorrect session data', () => {
+      of({ key: '' })
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(
+              new Error('Session Data is Incorrectly Defined')
+            ),
+        });
+      of({ key: undefined })
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          error: (err) =>
+            expect(err).toEqual(
+              new Error('Session Data is Incorrectly Defined')
+            ),
+        });
+    });
```

Comment:
Actually briefly looking at the assert function I just realized, would the application behave unexpectedly if we provided a blank session data object? i.e. If the admin who setup the `settings.json` missed out certain fields.

For e.g. If the bug reporting phase is open but the repo for that phase has not been defined.
```typescript
const settingJson = {
  openPhases: ['phaseBugReporting'],
  phaseTeamResponse: 'phaseTeamResponse'
};
```

Because the the part that checks for correct definition is simply this, doesn't seem to check if the fields are actually correct, simply that they are not blank.

```typescript
/**
 * Ensures that the input session Data has been correctly defined.
 * Returns true if satisfies these properties, false otherwise.
 * @param sessionData
 */
function isSessionDataCorrectlyDefined(sessionData: SessionData): boolean {
  for (const data of Object.values(sessionData)) {
    if (data === undefined || data === '') {
      return false;
    }
  }
  return true;
}
```

Cuz if this is the case then maybe we should open another PR to modify the part which checks for the correct sessionData Structure?

What do you think @anubh-v @dingyuchen ?

Reply:
Nice catch @ptvrajsk, yes we must verify that there is a repo defined for the open phase.
I'll create an issue for this.